File: Algorithm(2022)/BruteForcing/bruteForcing03.scala
package BruteForcing

object bruteForcing03 {
    def main(args: Array[String]): Unit = {
        val p = scala.io.StdIn.readLine()
        val row = p(1) - '0'
        val col = p(0) - 'a' + 1

        val dx = Array(-2, -2, -1, -1, 1, 1, 2, 2)
        val dy = Array(-1, 1, -2, 2, -2, 2, -1, 1)

        var cnt = 0

        for (i <- 0 until 8) {
            val nr = row + dx(i)
            val nc = col + dy(i)
            if (nr >= 1 && nc >= 1 && nr <= 8 && nc <= 8) cnt += 1
        }

        println(cnt)
    }
}

File: Algorithm(2022)/BruteForcing/bruteForcing03.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var p string
	fmt.Fscanln(reader, &p)

	row := int(p[1] - '0')
	col := int(p[0] - 'a' + 1)

	var dx = [8]int{-2, -2, -1, -1, 1, 1, 2, 2}
	var dy = [8]int{1, -1, 2, -2, -2, 2, -1, 1}

	cnt := 0

	for i := 0; i < 8; i++ {
		nr := row + dx[i]
		nc := col + dy[i]
		if nr >= 1 && nc >= 1 && nr <= 8 && nc <= 8 {
			cnt++
		}
	}
	fmt.Println(cnt)
}
